client/src/components/Users/Auth/SignUpCitizen/StepOne.tsx

```diff
@@ -0,0 +1,268 @@
+import React, { useState } from 'react';
+import { Box, Button, Checkbox, Typography, FormControlLabel } from '@mui/material';
+import { string, boolean, ValidationError, StringSchema, BooleanSchema } from 'yup';
+
+import StyledLink from '../../../StyledLink';
+import routes from '../../../../routes/routes';
+import PasswordInput from '../PasswordInput';
+import EmailInput from '../EmailInput';
+import NameInput from '../NameInput';
+
+import { useStyles } from './styles';
+
+const initialFormData = {
+  firstName: {
+    value: '',
+    error: null,
+    rule: string().required('Required.'),
+  },
+  lastName: {
+    value: '',
+    error: null,
+    rule: string().required('Required.'),
+  },
+  email: {
+    value: '',
+    error: null,
+    rule: string().email('Invalid email.').required('Required.'),
+  },
+  password: {
+    value: '',
+    error: null,
+    rule: string()
+      .min(8, 'Password is too short - should be 8 chars minimum.')
+      .required('Required.'),
+  },
+  passwordConfirm: {
+    value: '',
+    error: null,
+    rule: string()
+      .min(8, 'Password is too short - should be 8 chars minimum.')
+      .required('Required.'),
+  },
+  acceptTerms: {
+    value: false,
+    error: null,
+    rule: boolean().required('Required.'),
+  },
+};
+
+const getError = (rule: StringSchema | BooleanSchema, value: string | undefined) => {
+  try {
+    rule.validateSync(value);
+    return null;
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return error.message;
+    }
+  }
+};
+
+interface StepOneType {
+  initData: {};
+  handleNext: (formData: {}) => void;
+}
+
+export default function StepOne({ initData, handleNext }: StepOneType) {
+  const { classes } = useStyles();
+
+  Object.keys(initialFormData).forEach((key) => {
+    //@ts-ignore
+    initialFormData[key].value = initData[key];
```

Comment:
I've used `// @ts-ignore` on multiple lines as an escape hatch as I'm not sure how to get this to work with TypeScript.  If you guys have any ideas on how I could/should do this, that'd be awesome.